nextbest/src/main/java/no/difi/meldingsutveksling/nextbest/ConversationResource.java

```diff
@@ -0,0 +1,105 @@
+package no.difi.meldingsutveksling.nextbest;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.base.MoreObjects;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "direction")
+public abstract class ConversationResource {
+
+    @Id
+    private String conversationId;
+    private String senderId;
+    private String receiverId;
+    private String messagetypeId;
+    @JsonIgnore
+    private LocalDateTime lastUpdate;
+    @ElementCollection(fetch = FetchType.EAGER)
+    @MapKeyColumn(name = "fileid")
+    @Column(name = "filename")
+    @CollectionTable(name = "filerefs", joinColumns = @JoinColumn(name = "ids"))
+    private Map<Integer, String> fileRefs;
+
+    ConversationResource() {}
+
+    ConversationResource(String conversationId, String senderId, String receiverId, String messagetypeId,
+                         LocalDateTime lastUpdate, Map fileRefs){
+        this.conversationId = conversationId;
+        this.senderId = senderId;
+        this.receiverId = receiverId;
+        this.messagetypeId = messagetypeId;
+        this.lastUpdate = lastUpdate;
+        this.fileRefs = fileRefs;
+    }
+
+    public String getConversationId() {
+        return conversationId;
+    }
+
+    public void setConversationId(String conversationId) {
+        this.conversationId = conversationId;
+    }
+
+    public String getReceiverId() {
+        return receiverId;
+    }
+
+    public void setReceiverId(String receiverId) {
+        this.receiverId = receiverId;
+    }
+
+    public String getSenderId() {
+        return senderId;
+    }
+
+    public void setSenderId(String senderId) {
+        this.senderId = senderId;
+    }
+
+    public String getMessagetypeId() {
+        return messagetypeId;
+    }
+
+    public void setMessagetypeId(String messagetypeId) {
+        this.messagetypeId = messagetypeId;
+    }
+
+    public Map<Integer, String> getFileRefs() {
+        return fileRefs;
+    }
+
+    public LocalDateTime getLastUpdate() {
+        return lastUpdate;
+    }
+
+    public void setLastUpdate(LocalDateTime lastUpdate) {
+        this.lastUpdate = lastUpdate;
+    }
+
+    public void setFileRefs(HashMap<Integer, String> fileRefs) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "fileRefs" object should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1319)